pointcloud.js

```diff
@@ -455,232 +497,257 @@ proto.update = function(options) {
-        triOffset += 1
+      var x = points[i]
+      for(var j=0; j<3; ++j) {
+        upperBound[j] = Math.max(upperBound[j], x[j])
+        lowerBound[j] = Math.min(lowerBound[j], x[j])
       }
-    }
 
-    var cells = glyphLines.edges
-    var verts = glyphLines.positions
+      //Calculate text offset
+      if(alignment[0] < 0) {
+        textOffset[0] = alignment[0]  * (1+glyphBounds[1][0])
+      } else if(alignment[0] > 0) {
+        textOffset[0] = -alignment[0] * (1+glyphBounds[0][0])
+      }
 
-    for(var j=0; j<cells.length; ++j) {
-      var cell = cells[j]
-      for(var k=0; k<2; ++k) {
-        for(var l=0; l<3; ++l) {
-          positionArray[3*lineOffset+l] = x[l]
+      //Write out inner marker
+      var cells = glyphMesh.cells || []
+      var verts = glyphMesh.positions || []
+
+      for(var j=0; j<cells.length; ++j) {
+        var cell = cells[j]
+        for(var k=0; k<3; ++k) {
+          for(var l=0; l<3; ++l) {
+            positionArray[3*triOffset+l] = x[l]
+          }
+          for(var l=0; l<4; ++l) {
+            colorArray[4*triOffset+l] = color[l]
+          }
+          idArray[triOffset] = pickCounter
+          var p = verts[cell[k]]
+          glyphArray[2*triOffset]   = size * (cos*p[0] - sin*p[1] + textOffset[0])
+          glyphArray[2*triOffset+1] = size * (sin*p[0] + cos*p[1] + textOffset[1])
+          triOffset += 1
         }
-        for(var l=0; l<4; ++l) {
-          colorArray[4*lineOffset+l] = lineColor[l]
+      }
+
+      var cells = glyphLines.edges
+      var verts = glyphLines.positions
+
+      for(var j=0; j<cells.length; ++j) {
+        var cell = cells[j]
+        for(var k=0; k<2; ++k) {
+          for(var l=0; l<3; ++l) {
+            positionArray[3*lineOffset+l] = x[l]
+          }
+          for(var l=0; l<4; ++l) {
+            colorArray[4*lineOffset+l] = lineColor[l]
+          }
+          idArray[lineOffset] = pickCounter
+          var p = verts[cell[k]]
+          glyphArray[2*lineOffset]   = size * (cos*p[0] - sin*p[1] + textOffset[0])
+          glyphArray[2*lineOffset+1] = size * (sin*p[0] + cos*p[1] + textOffset[1])
+          lineOffset += 1
         }
-        idArray[lineOffset] = pickCounter
-        var p = verts[cell[k]]
-        glyphArray[2*lineOffset]   = size * (cos*p[0] - sin*p[1] + textOffset[0])
-        glyphArray[2*lineOffset+1] = size * (sin*p[0] + cos*p[1] + textOffset[1])
-        lineOffset += 1
       }
+
     }
 
-    //Increment pickCounter
-    pickCounter += 1
+
+
   }
 
+  //Update bounds
+  this.bounds = [lowerBound, upperBound]
+
+  //Save points
+  this.points = points
+
+  //Save number of points
+  this.pointCount = points.length
 
   //Update vertex counts
   this.vertexCount      = triVertexCount
   this.lineVertexCount  = lineVertexCount
 
-  //Update buffers
   this.pointBuffer.update(positionArray)
   this.colorBuffer.update(colorArray)
   this.glyphBuffer.update(glyphArray)
-  this.idBuffer.update(new Uint32Array(idArray))
```

Comment:
It seems we could simply pass the idArray while it is already declared as a Uint32 array.